backend/api/Services/DocumentService.cs

```diff
@@ -39,6 +49,45 @@ public IList<PimsActivityInstanceDocument> GetActivityDocuments(long activityId)
             return documentActivityRespostory.GetAll(activityId);
         }
 
+        public async Task<bool> DeleteActivityDocumentAsync(PimsActivityInstanceDocument activityDocument)
+        {
+            IList<PimsActivityInstanceDocument> existingActivityDocuments = documentActivityRespostory.GetAll(activityDocument.ActivityInstanceId);
+            if (existingActivityDocuments.Count == 1)
```

Comment:
Hmm, I don't think I understand this, we only try and delete the document if it is the only document associated with this activity?

Reply:
If there is only one document activity relationship, then remove it alongside the document and the stored document, otherwise only remove this relationship

Reply:
So if there are two different documents on an activity and we remove one, we will not delete that document even if that is the only instance of that document in the system? Won't that create an orphaned document in Mayan?

Reply:
Sorry @FuriousLlama either I'm not understanding this still or this is incorrect, but I want to resolve this before I approve.

Reply:
I see what you are saying. This was selected based on the activity side of the relationship. The changed one selects based on the document.